--- a/include/net/cfg80211.h
+++ b/include/net/cfg80211.h
@@ -3423,12 +3423,12 @@ enum wiphy_params_flags {
 #define IEEE80211_DEFAULT_AIRTIME_WEIGHT	256
 
 /* The per TXQ device queue limit in airtime */
-#define IEEE80211_DEFAULT_AQL_TXQ_LIMIT_L	5000
-#define IEEE80211_DEFAULT_AQL_TXQ_LIMIT_H	12000
+#define IEEE80211_DEFAULT_AQL_TXQ_LIMIT_L	1500
+#define IEEE80211_DEFAULT_AQL_TXQ_LIMIT_H	5000
 #define IEEE80211_DEFAULT_AQL_TXQ_LIMIT_BC	50000
 
 /* The per interface airtime threshold to switch to lower queue limit */
-#define IEEE80211_AQL_THRESHOLD			24000
+#define IEEE80211_AQL_THRESHOLD			12000
 
 /**
  * struct cfg80211_pmksa - PMK Security Association
